File: 2264-largest-3-same-digit-number-in-string/2264-largest-3-same-digit-number-in-string.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        n = len(num)
        max_good = ""
        for i in range(n - 2):
            sub = num[i:i+3]
            if sub[0] == sub[1] == sub[2]:
                if sub > max_good:
                    max_good = sub
        return max_good

File: 2264-largest-3-same-digit-number-in-string/2264-largest-3-same-digit-number-in-string.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
        string max_good = "";
    for (int i = 0; i + 2 < (int)num.size(); ++i) {
        // 取長度為 3 的 substring
        string sub = num.substr(i, 3);
        // 判斷三位是否相同
        if (sub[0] == sub[1] && sub[1] == sub[2]) {
            if (sub > max_good) {
                max_good = sub;
            }
        }
    }
    return max_good;
    }
};